design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,679 @@
+proposed Template Stack Manager (and approaches to overcome them) are provided.
+
+### Out of Scope
+
+Throughout the design, Go templates are used to illustrate potential usage, but
+this is an implementation detail that will be the subject of additional design
+documents.
+
+This design is also not overly concerned with the format of the `Stack` object
+or the layout of the Stack filesystem. Those topics will be discussed in another
+one-pager.
+
+Finally, the specific resources that a Template Stack may render is not in scope
+for this design. There is some thinking on this included in [Concerning the use
+of Core Types](#Concerning-the-use-of-Core-Types).
+
+## Benefits of Template Stacks
+
+The current design of Stacks allow for a new CRD to be managed by an independent
+controller. A CR will be handled by the Stack controller in any way that
+controller sees fit. This includes issuing remote API calls for additional
+inputs or creating external outputs, accessing local resources, or performing
+complex computations.
+
+Stacks may also opt to take advantage of the Crossplane Runtime to simplify the
+life cycle management of resources.
+
+In contrast, this proposal introduces Template Stacks with the limited ability
+to use their CR and child resources as input to create, update, or delete child
+Kubernetes resources. Application owners and managers can take advantage of
+Template Stacks to package complex deployments, while users benefit from a
+simple install and usage interface. Developers can use Template Stacks as
+application owners would, to create Kubernetes resource machinery using template
+engine logic, circumventing the need more complex controllers.
+
+The Template Stack Manager will permit the use of template variables to create
+and manage a set of resources provided as a set of strings included in the
+definition of the Stack.
+
+The benefits of Template Stacks are as follows:
+
+* [Simple Variable Substitution](#Simple-Variable-Substitution)
+
+  Stack Template resources can take input from a CR spec or from sibling
+  resources.
+* [No Custom Controller Needed](#No-Custom-Controller-Needed)
+
+  There is a single all-purpose controller that is designed to managed Template
+  Stacks. The existing Stack Manager will wire the Template Stack Manager for
+  managing the resources defined in a Template Stack.
+* [Composition](#Composition)
+
+  Stacks can be embedded in different Cluster environments, including other
+  Stacks. Charts and custom YAML files may reside in the cluster in the form of
+  a Template Stack.
+
+  Template Stacks introduce template variables that can take advantage of the
+  field values from other template rendered resources. Template variable
+  substitution is performed in each Kubernetes reconciliation pass, affected
+  all template resources for a given Stack CRD.
+
+### Simple Variable Substitution
+
+If an application owner wanted to parameterize a single `Deployment` within a
```

Comment:
I was trying to start with a resource that may be more generally understood.  I could replace `Deployment` with any arbitrary kind, or argue that the missing `apiVersion` on `Deployment` means that this could be a `deployment.crossplane.example.org/v1alpha1` :smile_cat: 

I'll replace it with a silly example type that has less pretext. 

```
kind: Coffee
spec:
  creamer: {{ .creamer }}
```
[additives?](https://arstechnica.com/civis/viewtopic.php?t=82197)